src/tribler/core/user_activity/manager.py

```diff
@@ -41,12 +41,11 @@ def __init__(self, task_manager: TaskManager, session: Session, max_query_histor
         self.task_manager.register_task("Check preferable", self.check_preferable,
                                         interval=session.config.get("user_activity/health_check_interval"))
 
-    def on_query_results(self, data: dict) -> None:
```

Comment:
If `query` is always a `str`, the next `if query is None:` can be removed.

Reply:
(this comment should be on the addition, not on the removal - woops)

Reply:
I basically made the typing up, thinking a search query without a query didn't make sense. Having said that, I can't really guarantee that's not somehow magically needed. The check was there before.
So, I'll either fix the typing or remove the check. Which do you prefer?

Reply:
Like you said, I think it is sensible to change this to a strict `str`. For this reason, I'd prefer removing the check.

Reply:
Done. There were even tests for it, which I also removed.